src/sqlfluff/core/rules/base.py

```diff
@@ -555,39 +555,28 @@ def crawl(
             )
             return vs, raw_stack, fixes, memory
 
-        new_lerrs = []
-        new_fixes = []
-
-        def _process_lint_result(res):
-            self.discard_unsafe_fixes(res, templated_file)
-            lerr = res.to_linting_error(rule=self)
-            ignored = False
-            if lerr:
-                if ignore_mask:
-                    filtered = LintedFile.ignore_masked_violations([lerr], ignore_mask)
-                    if not filtered:
-                        lerr = None
-                        ignored = True
-            if lerr:
-                new_lerrs.append(lerr)
-            if not ignored:
-                new_fixes.extend(res.fixes)
```

Comment:
I moved the nested function `_process_lint_result()` to a separate method on `RuleBase` so the automated test can use it.